scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/task/internal/InternalTask.java

```diff
@@ -540,6 +557,50 @@ public synchronized void update(TaskInfo taskInfo) {
         this.taskInfo = (TaskInfoImpl) taskInfo;
     }
 
+    /**
+     * Loads authorized selection scripts.
+     */
+    private static void loadAuthorizedScriptsSignatures(TaskId id) {
+        String dirName = PASchedulerProperties.EXECUTE_SCRIPT_AUTHORIZED_DIR.getValueAsString();
+        if (dirName != null && dirName.length() > 0) {
+            dirName = PASchedulerProperties.getAbsolutePath(dirName);
+            File folder = new File(dirName);
+
+            if (folder.exists() && folder.isDirectory()) {
+                logger.info(id, "The resource manager will accept only fork environment or cleaning scripts from " + dirName);
+                long currentTime = System.currentTimeMillis();
+                long configuredAuthorizedScriptLoadPeriod = DEFAULT_AUTHORIZED_SCRIPT_LOAD_PERIOD;
+                if (PASchedulerProperties.EXECUTE_SCRIPT_AUTHORIZED_DIR_REFRESHPERIOD.isSet()) {
+                    configuredAuthorizedScriptLoadPeriod = PASchedulerProperties.EXECUTE_SCRIPT_AUTHORIZED_DIR_REFRESHPERIOD.getValueAsInt();
+                }
+                if (currentTime - lastAuthorizedFolderLoadingTime > configuredAuthorizedScriptLoadPeriod) {
+                    lastAuthorizedFolderLoadingTime = currentTime;
+                    authorizedSelectionScripts = new HashSet<>();
+                    for (File file : folder.listFiles()) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Refactor this code to not nest more than 3 if/for/while/switch/try statements. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS134)